lib/FilesHooks.php

```diff
@@ -499,7 +513,30 @@ protected function generateMoveActivities($users, $beforePathMap, $afterPathMap,
 	 * @return array
 	 */
 	protected function getUserPathsFromPath($path, $uidOwner) {
-		return Share::getUsersSharingFile($path, $uidOwner, true, true);
+		try {
+			$node = $this->rootFolder->getUserFolder($uidOwner)->get($path);
+		} catch (NotFoundException $e) {
+			return [];
+		}
+
+		if (!$node instanceof Node) {
+			return [];
+		}
+		$accessList = $this->shareManager->getAccessList($node, true, true);
+		$helper = new ShareHelper($this->rootFolder); // FIXME
+
+		$nodes = $helper->getPathsForAccessList($node, $accessList['users']);
```

Comment:
I don't think memory issues are the main concern here. Yes it is slow. But as discussed the old endpoint just returned false info in some cases.

Would it maybe make sense to just use the filename and link to the file by using not a path but the nodeid?